src/widgets/immunizations/immunizations-form.component.tsx

```diff
@@ -0,0 +1,348 @@
+
+export function ImmunizationsForm(props: ImmunizationsFormProps) {
+  const [vaccineName, setVaccineName] = useState("");
+  const [vaccineUuid, setVaccineUuid] = useState("");
+  const [immunizationObsUuid, setImmunizationObsUuid] = useState("");
+  const [vaccinationDate, setVaccinationDate] = useState(null);
+  const [immunizationSequences, setImmunizationSequences] = useState([]);
+  const [currentDose, setCurrentDose] = useState<ImmunizationSequence>(
+    {} as ImmunizationSequence
+  );
+  const [vaccinationExpiration, setVaccinationExpiration] = useState(null);
+  const [lotNumber, setLotNumber] = useState("");
+  const [manufacturer, setManufacturer] = useState("");
+  const [enableCreateButtons, setEnableCreateButtons] = useState(false);
+  const [enableEditButtons, setEnableEditButtons] = useState(true);
+  const [viewEditForm, setViewEditForm] = useState(true);
+  const [formChanged, setFormChanged] = useState<Boolean>(false);
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+  const formRef = useRef<HTMLFormElement>(null);
+  const { t } = useTranslation();
+  const history = useHistory();
+  const today = new Date().toISOString().split("T")[0];
+  const currentUser = useSessionUser();
+
+  useEffect(() => {
+    if (vaccinationDate) {
+      setEnableCreateButtons(true);
+    } else {
+      setEnableCreateButtons(false);
+    }
+  }, [vaccinationDate]);
+
+  useEffect(() => {
+    if (viewEditForm && formChanged) {
+      setEnableEditButtons(true);
+    } else {
+      setEnableEditButtons(false);
+    }
+  }, [viewEditForm, formChanged]);
+
+  useEffect(() => {
+    if (props.match.params) {
+      const {
+        immunizationObsUuid,
+        vaccineName,
+        vaccineUuid,
+        manufacturer,
+        expirationDate,
+        vaccinationDate,
+        lotNumber,
+        sequences,
+        currentDose
+      }: Immunization = props.match.params[0];
+
+      setImmunizationObsUuid(immunizationObsUuid);
+      setVaccineName(vaccineName);
+      setVaccineUuid(vaccineUuid);
+      setManufacturer(manufacturer);
+      setVaccinationDate(vaccinationDate);
+      setVaccinationExpiration(expirationDate);
+      setLotNumber(lotNumber);
+      setViewEditForm(false);
+      if (hasSequences(sequences)) {
+        setImmunizationSequences(sequences);
+      }
+      if (vaccineName && vaccinationDate) {
+        setViewEditForm(true);
+        if (hasSequences(sequences)) {
+          setCurrentDose(currentDose);
+        }
+      }
+    }
+  }, [props.match.params]);
+
+  const handleFormSubmit = event => {
+    event.preventDefault();
+    const currentVisitUuid = getStartedVisit?.getValue()?.visitData?.uuid;
+    const currentLocationUuid = currentUser?.sessionLocation?.uuid;
+    const currentProviderUuid = currentUser?.currentProvider?.uuid;
+
+    const immunization: Immunization = {
+      patientUuid: patientUuid,
```

Comment:
Please use shorthands if possible

Reply:
Sure, that makes sense! I will change this bit.